javascript/ql/lib/semmle/javascript/Concepts.qll

```diff
@@ -84,6 +84,17 @@ abstract class FileNameSource extends DataFlow::Node { }
 abstract class DatabaseAccess extends DataFlow::Node {
   /** Gets an argument to this database access that is interpreted as a query. */
   abstract DataFlow::Node getAQueryArgument();
+
+  /** Gets a node to which a result of the access may flow. */
+  abstract DataFlow::Node getAResult();
```

Comment:
```suggestion
  DataFlow::Node getAResult() {
    none() // Overridden in subclass
  }
```

We try to keep backwards comparability with QL code written by others.  
Adding a new abstract predicate will result in a compilation error for any existing implementation. 

An alternative pattern we often use is to add the predicate with a default `none()` implementation.

Reply:
perfect, will do